File: Ruby/Selection-Sort.rb
def is_sorted?(arr)
    len = arr.length()

    0.upto(len-2) do |x|
        if arr[x] > arr[x+1]
            return false
        end
    end

    return true
end


def selection_sort(arr)
    len = arr.length

    0.upto(len-2) do |i|
        min = i
        (i+1).upto(len-1) do |j|
            if arr[min] > arr[j]
                min = j
            end
        end
        arr[i], arr[min] = arr[min], arr[i]
    end

    return arr
end


if __FILE__ == $0
    arr = 10.times.map { |x| (Random.rand() * 100).to_i + x }
    puts "Array #{arr}. Sorted? #{is_sorted?(arr)}"
    arr = selection_sort(arr)
    puts "Array #{arr}. Sorted? #{is_sorted?(arr)}"
end

File: Python/Selection-Sort.py
from random import randint


def is_sorted(arr):
    for x in range(len(arr)-1):
        if arr[x] > arr[x+1]:
            return False
    return True


def selection_sort(arr):
    minimum = 0
    length = len(arr)

    for i in range(length-1):
        minimum = i

        for j in range(i+1, length):
            if arr[minimum] > arr[j]:
                minimum = j

        arr[i], arr[minimum] = arr[minimum], arr[i]

    return arr


if __name__ == '__main__':
    arr = [randint(0, 100) for _ in range(10)]
    print(f'Array {arr}. Sorted? {is_sorted(arr)}')
    arr = selection_sort(arr)
    print(f'Array {arr}. Sorted? {is_sorted(arr)}')

